src/app/examples/list-load-more-example/list-load-more-example.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component } from '@angular/core';
+
+import { BaseListComponent } from '../list-example/base-list.component';
+import { KirbyLoadMoreEvent } from './../../../kirby/components/list/list.event';
+
+@Component({
+  selector: 'kirby-list-load-more-example',
+  templateUrl: './list-load-more-example.component.html',
+  styleUrls: ['./list-load-more-example.component.scss'],
+})
+export class ListLoadMoreExampleComponent extends BaseListComponent {
+  private itemCount: number = 0;
+
+  constructor() {
+    super();
+    this.items.push(...this.generateItems());
+  }
+
+  onLoadMore(onLoadMoreEvent: KirbyLoadMoreEvent): void {
+    // We end the load more event after 20 items, by sending null to the kirby list.
+    if (this.itemCount <= 20) {
+      // lets make a delay to simulate a HTTP call.
+      new Promise((resolve) => setTimeout(resolve, 2000)).then(() => {
```

Comment:
Simplify:
```suggestion
      setTimeout(() => {
        this.items.push(...this.generateItems());
        onLoadMoreEvent.complete();
      }, 2000);
```
OR if you want to use an `Observable` (as the httpclient interface):
```suggestion
      of(this.generateItems()).pipe(delay(2000))....
```